Bio/Seq.py

```diff
@@ -2590,8 +2590,8 @@ def _translate_str(
     amino_acids = []
     forward_table = table.forward_table
     stop_codons = table.stop_codons
-    if table.nucleotide_alphabet.letters is not None:
-        valid_letters = set(table.nucleotide_alphabet.letters.upper())
+    if table._is_dna:
+        valid_letters = set(_ambiguous_dna_letters.upper())
```

Comment:
I think this function is broken with release 1.77 and earlier:
```
>>> from Bio.Seq import translate
>>> translate("TGA") # stop codon
'*'
>>> translate("TGN") # possible stop codon
'X'
>>> translate("ANT") # definitely not a stop codon
'X'
```
where `'X'` stands for a possible stop codon such as `'TAN'` or `'NNN'`.

Instead of `table._is_dna` it may be better to get the `valid_letters` directly from the `table`. But it depends how we solve the bug above.

Reply:
What do you mean by broken? ``'X'`` or ``'Xaa'`` stands for an unknown amino acid. In some cases we're using it for entries which might even be a stop codon.

Reply:
I am wondering about the line
```
            elif valid_letters.issuperset(set(codon)):
                # Possible stop codon (e.g. NNN or TAN)
                amino_acids.append(pos_stop)
```
in `_translate_str` called by this function. `set(codon)` will ignore the order of nucleotides, so a possible stop codon `TAN` is treated the same way as `ANT` which is not a possible stop codon.

Reply:
That does look odd...

Reply:
```python
>>> from Bio.Seq import _translate_str
>>> from Bio.Data.CodonTable import ambiguous_generic_by_id as tables
>>> t("TAA", table=tables[1], pos_stop="@")
'*'
>>> t("TAN", table=tables[1], pos_stop="@")
'@'
>>> t("ANT", table=tables[1], pos_stop="@")
'X'
```

I don't think examples like ``ANT`` never reach this if statement, they are handled give the forward table.